iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOManager.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.openhouse.internal.catalog.fileio;
+
+import static com.linkedin.openhouse.cluster.storage.StorageType.HDFS;
+import static com.linkedin.openhouse.cluster.storage.StorageType.LOCAL;
+
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.io.FileIO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the main class that provides the FileIO implementation based on the storage type. Each
+ * storage type should have a corresponding FileIO bean field defined in this class and the
+ * corresponding FileIO bean should be returned for appropriate storage type in the method {@link
+ * #getFileIO(StorageType.Type)}. If the storage type is not configured, the method should throw an
+ * IllegalArgumentException.
+ */
+@Component
+public class FileIOManager {
+
+  @Autowired(required = false)
+  @Qualifier("HadoopFileIO")
```

Comment:
we add a qualifier to this instance of HadoopFileIO so that it doesn't conflict with FileIO already present from `FsStorageProvider`